### DISCUSSION TITLE
Some questions about multiple port extensions

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#19330] [Q&A]
Hello,

I've observed that Tasmota supports up to 4 PCF8574 modules. Is there a possibility to raise this limit?

Is the PCF8575 (16 IO ports) still not supported?

How does the limit of 4 PCF8574 modules interact when PCA9685 modules are added additionally? Does the limit of 4 still apply to each of them?

Thank you!

**Comment by lalo-uy** [comment]
PCF8574 Only have 4 possible addresses ( 2 addr pins) so no more than 4 in
the same bus.

**Comment by arendst** [comment]
PCF8574 has three address pins. Same for PCF8574A. So in theory 16 chips can be addressed.

Tasmota Mode 1 supports storing settings for up to four chips hence the restriction of max four chips. Mode 2 could support more chips but the current implementation sticks to four.

If you need more I/O consider using MCP230xx.
